src/fetchClient.ts

```diff
@@ -53,7 +53,8 @@ export class FetchClient {
   }
 
   private static async returnData<T>(
-    response: Response
+    response: Response,
+    sanitizer?: (data: Partial<T>) => T
```

Comment:
```suggestion
    sanitizer?: DataSanitizer<T>
```
I think you already have a nice type created for this that you can reuse